static_table.go

```diff
@@ -101,3 +101,133 @@ var staticTableEntries = [...]HeaderField{
 	{Name: "x-frame-options", Value: "deny"},
 	{Name: "x-frame-options", Value: "sameorigin"},
 }
+
+type indexAndValues struct {
+	idx    int
+	values map[string]int
+}
+
+// A map of the header names from the static table to their index in the table.
+// This is used by the encoder to quickly find if a header is in the static table
+// and what value should be used to encode it.
+// There's a second level of mapping for the headers that have some predefined
+// values in the static table.
+var encoderMap = map[string]indexAndValues{
```

Comment:
Can you add a unit test that checks that encoderMap and staticTableEntries are consistent, e.g. by looping over one of them and checking that the indexes match (and the entry exists)?

Reply:
Done.